Include/internal/pycore_frame.h

```diff
@@ -58,26 +58,21 @@ typedef struct _PyInterpreterFrame {
     PyObject *f_builtins; /* Borrowed reference. Only valid if not on C stack */
     PyObject *f_locals; /* Strong reference, may be NULL. Only valid if not on C stack */
     PyFrameObject *frame_obj; /* Strong reference, may be NULL. Only valid if not on C stack */
-    // NOTE: This is not necessarily the last instruction started in the given
-    // frame. Rather, it is the code unit *prior to* the *next* instruction. For
-    // example, it may be an inline CACHE entry, an instruction we just jumped
-    // over, or (in the case of a newly-created frame) a totally invalid value:
-    _Py_CODEUNIT *prev_instr;
+    /* The instruction that is currently executing (possibly not started yet). */
+    _Py_CODEUNIT *instr_ptr;
     int stacktop;  /* Offset of TOS from localsplus  */
-    /* The return_offset determines where a `RETURN` should go in the caller,
-     * relative to `prev_instr`.
-     * It is only meaningful to the callee,
-     * so it needs to be set in any CALL (to a Python function)
-     * or SEND (to a coroutine or generator).
-     * If there is no callee, then it is meaningless. */
-    uint16_t return_offset;
+    /* The next_instr_offset determines where the next instruction is relative
+     * to instr_ptr. It enables us to keep instr_ptr pointing to the current
+     * instruction until it is time to begin executing the next one, which is
+     * necessary for tracebacks and tracing. */
+    uint16_t next_instr_offset;
```

Comment:
The reason I didn't do it that way was because then the line instrumentation doesn't know what the previous instruction is (the instr_ptr is already updated when we get there).  We would need another way to make line instrumentation work.

Reply:
> I much prefer keeping return_offset, which only needs to be set for calls and sends.

We need to use it now in YIELD as well as in RETURN, so it might need to be renamed anyway.

Reply:
> The reason I didn't do it that way was because then the line instrumentation doesn't know what the previous instruction is (the instr_ptr is already updated when we get there). We would need another way to make line instrumentation work.

I pushed a change where I use `frame->instr_ptr - frame->next_instr_offset` to find the previous instruction. But this requires `frame->next_instr_offset` to always be `0` when it is not valid.